src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala

```diff
@@ -362,3 +404,33 @@ class DeltaCatalog(val spark: SparkSession) extends DelegatingCatalogExtension
     loadTable(ident)
   }
 }
+
+case class OtherFormatsStagedTable(
+    delegateCatalog: TableCatalog,
+    ident: Identifier,
+    schema: StructType,
+    partitions: Array[Transform],
+    override val properties: util.Map[String, String]) extends StagedTable with SupportsWrite {
+
+  private lazy val realTable = delegateCatalog.createTable(ident, schema, partitions, properties)
+
+  override def capabilities(): util.Set[TableCapability] = realTable.capabilities()
+
+  override def name(): String = realTable.name()
+
+  override def commitStagedChanges(): Unit = {}
+
+  override def abortStagedChanges(): Unit = {
+    try delegateCatalog.dropTable(ident) catch {
```

Comment:
I'm not sure it's right to just drop the table. What if it was being RTASed?

Reply:
RTAS also drops the table for non-atomic tables. It is documented that this is inherently a very dangerous operation